core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/DefaultHealthCheckerContext.java

```diff
@@ -48,6 +49,11 @@
 final class DefaultHealthCheckerContext
         extends AbstractExecutorService implements HealthCheckerContext, ScheduledExecutorService {
 
+    /**
+     * Lock for {@link scheduledFutures}.
```

Comment:
```suggestion
     * Lock for {@code scheduledFutures}.
```

Reply:
How about putting this together with `scheduledFutures`?

Reply:
Would be worth using `GuardedBy` for better static analaysis? https://errorprone.info/bugpattern/GuardedBy

Reply:
Let's use that. 😄